MantidQt/CustomInterfaces/test/ReflDataProcessorPresenterTest.h

```diff
@@ -0,0 +1,524 @@
+          << dqq << scale << options;
+    }
+    return ws;
+  }
+
+  ITableWorkspace_sptr
+  createPrefilledMixedWorkspace(const std::string &wsName,
+                                const DataProcessorWhiteList &whitelist) {
+    auto ws = createWorkspace(wsName, whitelist);
+    const std::string group = "0";
+    const std::vector<std::string> run{"38415", "38417"};
+    const std::string angle = "0.5069";
+    const std::string transRun = "38393";
+    const std::string qMin = "0.0065";
+    const std::string qMax = "0.0737";
+    const std::vector<std::string> dqq{"0.0148", "0.0198"};
+    const std::string scale = "1";
+    const std::string options = "";
+    for (int i = 0; i < 2; ++i) {
+      TableRow row = ws->appendRow();
+      row << group << run[i] << angle << transRun << qMin << qMax << dqq[i]
+          << scale << options;
+    }
+    return ws;
+  }
+
+  ITableWorkspace_sptr
+  createPrefilledMinimalWorkspace(const std::string &wsName,
+                                  const DataProcessorWhiteList &whitelist) {
+
+    auto ws = createWorkspace(wsName, whitelist);
+    const std::string group = "0";
+    const std::string run = "38415";
+    const std::string angle = "0.5069";
+    const std::string transRun = "";
+    const std::string qMin = "0.0065";
+    const std::string qMax = "0.0737";
+    const std::string dqq = "0.0148";
+    const std::string scale = "1";
+    const std::string options = "";
+    TableRow row = ws->appendRow();
+    row << group << run << angle << transRun << qMin << qMax << dqq << scale
+        << options;
+    return ws;
+  }
+
+  void createSampleEventWS(const std::string &wsName) {
+    auto tinyWS = WorkspaceCreationHelper::createEventWorkspace2();
+    AnalysisDataService::Instance().addOrReplace(wsName, tinyWS);
+  }
+
+  ReflGenericDataProcessorPresenterFactory presenterFactory;
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static ReflDataProcessorPresenterTest *createSuite() {
+    return new ReflDataProcessorPresenterTest();
+  }
+  static void destroySuite(ReflDataProcessorPresenterTest *suite) {
+    delete suite;
+  }
+
+  ReflDataProcessorPresenterTest() { FrameworkManager::Instance(); }
+
+  void testProcessEventWorkspacesCustomSlicing() {
+    NiceMock<MockDataProcessorView> mockDataProcessorView;
+    NiceMock<MockProgressableView> mockProgress;
+    NiceMock<MockMainPresenter> mockMainPresenter;
+    auto presenter = presenterFactory.create();
+    presenter->acceptViews(&mockDataProcessorView, &mockProgress);
+    presenter->accept(&mockMainPresenter);
+
+    createPrefilledWorkspace("TestWorkspace", presenter->getWhiteList());
+    EXPECT_CALL(mockDataProcessorView, getWorkspaceToOpen())
+        .Times(1)
+        .WillRepeatedly(Return("TestWorkspace"));
+    presenter->notify(DataProcessorPresenter::OpenTableFlag);
```

Comment:
I would assert something:
`TS_ASSERT_THROWS_NOTHING(presenter->notify(DataProcessorPresenter::OpenTableFlag));`
(for all presenter->notify ?)